### PR TITLE
Improve pkg-config file generation

### PR DESCRIPTION
**Pull request by kmilos**

### CODE REVIEW

File: cmake/exiv2.pc.in [hunk 1]
```diff
@@ -8,4 +8,5 @@ Description: @PROJECT_DESCRIPTION@
 Version: @PROJECT_VERSION@
 URL: @PACKAGE_URL@
 Libs: -L${libdir} -lexiv2
+Libs.private: -linih
```
Comment (kmilos) [inline cmake/exiv2.pc.in:11]: Actually, this needs to depend on EXIV2_ENABLE_INIH...

**Comment by neheb** [comment]
my meson PR produces

```
prefix=/usr/local
includedir=${prefix}/include
libdir=${prefix}/lib64

Name: exiv2
Description: Exif/IPTC/Xmp C++ metadata library and tools plus ICC Profiles, Previews and more.
URL: https://exiv2.org
Version: 1.0.0
Requires.private: INIReader, libbrotlidec, libcurl, zlib
Libs: -L${libdir} -lexiv2
Cflags: -I${includedir}
```

I assume it's similar here?

**Comment by kmilos** [comment]
Thanks for comapring and letting me know @neheb 

Indeed, there are further two things I'll need to fix: no need fir further dependencies (`inih` and `libbrotlicommon`; they'll get pulled in by the respective .pc files), and comma delimited list instead of space delimited.

I'm also surprised I don't see `iconv` in your list...

**Comment by kmilos** [comment]
I am now getting (on Windows, MSYS2 UCRT64):
```
prefix=/ucrt64
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: exiv2
Description: Exif/IPTC/Xmp C++ metadata library and tools plus ICC Profiles, Previews and more.
Version: 1.0.0.9
URL: https://exiv2.org
Libs: -L${libdir} -lexiv2
Libs.private: -lintl
Cflags: -I${includedir}
Requires.private: INIReader, expat, iconv, libbrotlidec, libcurl, zlib
```

**Comment by neheb** [comment]
Iconv is libc iconv. Meson does not need to add. I’ll check expat.

**Comment by kmilos** [comment]
Luckily we require CMake 3.11 anyway, so removed our custom module: https://cmake.org/cmake/help/latest/module/FindIconv.html

**Comment by neheb** [comment]
Expat is not implemented in meson as it’s simpler. I might look into this.

**Review by piponazo — DISMISSED** [review]

**Comment by neheb** [comment]
Windows:

```
prefix=/usr/local
includedir=${prefix}/include
libdir=${prefix}/lib

Name: exiv2
Description: Exif/IPTC/Xmp C++ metadata library and tools plus ICC Profiles, Previews and more.
URL: https://exiv2.org
Version: 1.0.0
Requires.private: INIReader, libbrotlidec, libcurl, zlib
Libs: -L${libdir} -lexiv2
Libs.private: -lws2_32 -liconv -lintl
Cflags: -I${includedir}
```

**Comment by kmilos** [comment]
Hm, I have iconv.pc on MSYS2...

**Comment by neheb** [comment]
On meson I check for it no matter the OS. Probably non Windows with CMake.

**Comment by kmilos** [comment]
Yep, MSYS2 was an exception rather than the rule, so converted iconv to Libs.private as well now.

**Review by neheb — APPROVED** [review]

**Comment by eli-schwartz** [comment]
> Expat is not implemented in meson as it’s simpler. I might look into this.

expat provides a canonical pkg-config file so there's no need for Meson to implement an override.

**Comment by neheb** [comment]
no I meant in the PR. I've since implemented it.

**Pull request merged**